src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/CustomItemEffects.java

```diff
@@ -852,37 +932,72 @@ public void renderBlockOverlay(DrawBlockHighlightEvent event) {
 								HashSet<BlockPos> candidatesOld = new HashSet<>();
 								TreeMap<Float, Set<BlockPos>> candidatesOldSorted = new TreeMap<>();
 
-								getBuildersWandCandidates(
-									player,
-									event.target,
-									event.partialTicks,
-									candidatesOld,
-									candidatesOldSorted,
-									10
-								);
+								boolean ruler = heldInternal.equals("BUILDERS_RULER");
 
-								ItemStack matchStack = new ItemStack(matchItem, 1,
-									match.getBlock().getDamageValue(Minecraft.getMinecraft().theWorld, event.target.getBlockPos())
-								);
+								if (!ruler) {
+									getBuildersWandCandidates(
+										player,
+										event.target,
+										event.partialTicks,
+										candidatesOld,
+										candidatesOldSorted,
+										10
+									);
+								} else {
+									getBuildersRulerCandidates(
+										player,
+										event.target,
+										event.partialTicks,
+										candidatesOld,
+										candidatesOldSorted,
```

Comment:
holy shit, this is literally half my screen just in indentation

Reply:
also again, this is a programming language which does have functions, consider making use of them